### ISSUE TITLE
rewrite rules for scala 3 breaks compilation

### ISSUE THREAD

**Issue by mlachkar-da** [scalameta/scalafmt#3337]
## Configuration (required) ##

Please paste the contents of your `.scalafmt.conf` file here:
```
version = 3.5.9

project.git = true

align.preset = none # never align to make the fmt more diff friendly
maxColumn = 100
runner.fatalWarnings = true
trailingCommas = multiple
# Disable scala doc wrapping (behavior changed in v3.0.0).
docstrings.wrap = no
newlines.topLevelStatementBlankLines = [
  { blanks { after = 1 }, regex = "^Import" }
]

rewrite.scala3.convertToNewSyntax = true
runner.dialect = "Scala213Source3"

...
```

NB: before submitting, please confirm that the problem is observed in the
*latest published* version of the formatter! We don't publish hotfixes for older
versions, and the problem you have observed in an older version may have already
been fixed.

## Command-line parameters (required) ##

When I run scalafmt via CLI like this: `<command-line parameter>`

If you were not using CLI initially:
- please check using `scalafmt` CLI before submitting here (required)
- if the problem does not present via CLI, DO NOT SUBMIT
  here: this is NOT a formatter issue
  - it could be an issue with your integration (sbt-scalafmt, metals, intellij etc.)
  - instead please submit the issue in the appropriate tool's repository
  - for instance, if the tool is `sbt` with the `sbt-scalafmt` plugin: please
    submit in `sbt-scalafmt` and include scalafmt parameters from `build.sbt`
    and the `sbt` command used
  - conversely, do not submit formatting issues into other tools; you must first
    verify using `scalafmt` CLI

## Steps

Given code like this:
```scala
    Lifecycle.close(
      tokenProvider +:
        channelPerEndpoint.toList.map { case (endpoint, channel) =>
          new CloseableChannel(channel, logger, s"grpc-client-auth-$endpoint")
        }: _*
    )(logger)
```

## Problem

Scalafmt formats code like this:
```scala
    Lifecycle.close(
      tokenProvider +:
        channelPerEndpoint.toList.map { case (endpoint, channel) =>
          new CloseableChannel(channel, logger, s"grpc-client-auth-$endpoint")
        }*
    )(logger)
```

## Expectation
I expect the code to compile.

**Comment by mlachkar-da** [comment]
this syntax is not yet allowed on scala213source3 dialect.

**Comment by tgodzik** [comment]
That's weird, I thought it's supported based on https://github.com/scala/scala/pull/9584

**Comment by tgodzik** [comment]
It does compile for me :thinking:

**Comment by mlachkar-da** [comment]
you're right! 
let me check which scala version I was using

**Comment by tgodzik** [comment]
Looks like it should get rewritten to `val aaa = test(("" +: Seq("").map{_ + ""})*)` with the added parenthesis, but looks like a bug in the compiler.

**Comment by mlachkar-da** [comment]
I can do a workaround for this, by extracting the value first.

**Comment by tgodzik** [comment]
That might be a good workaround for now, though we should fix it in scalafmt. I think it might have soemthing to do with operator precedence :thinking:

**Comment by kitbellew** [comment]
scalafmt will not *add* the parens here, so what's the right solution? not to rewrite this case, keep `: _`?

**Comment by mlachkar-da** [comment]
I think it either needs to add the parenthesis, which I think should be the best solution, or at least do nothing.

**Comment by kitbellew** [comment]
i don't think we have a way (yet) to add anything. we can remove or replace :) i had some code in the works a long time ago, but didn't finish it...
